packages/firestore/src/util/promise.ts

```diff
@@ -23,7 +23,8 @@ export interface Rejecter {
   (reason?: Error): void;
 }
 
-export interface CancelablePromise<T> extends Promise<T> {
+export interface CancelablePromise<T>
+  extends Pick<Promise<T>, Exclude<keyof Promise<T>, 'finally'>> {
```

Comment:
This can be simplified to:

```suggestion
  extends Omit<Promise<T>, 'finally'> {
```

It would probably improve readability.